cmd/oras/internal/option/packer.go

```diff
@@ -77,6 +77,9 @@ func (opts *Packer) LoadManifestAnnotations() (annotations map[string]map[string
 		if err = decodeJSON(opts.AnnotationFilePath, &annotations); err != nil {
 			return nil, err
 		}
+		if len(annotations[annotationManifest]) == 0 && len(opts.FileRefs) == 0 {
+			return nil, fmt.Errorf("no blob and manifest annotation file %s is empty", opts.AnnotationFilePath)
+		}
```

Comment:
This check overlaps with `validateEmpty()` if it is put after loading all the annotations.

Reply:
`validateEmpty()` will return error only when file reference, `AnnotationFilePath` tag and `ManifestAnnotations` tag are not provided at the same time. However, here it checks for the condition when `AnnotationFilePath `  is not empty but it have a content like `{"{"$manifest":{}}"}` and no file reference provided.